### PR TITLE
Implement register_phone_number

### PR DESCRIPTION
**Pull request by megrogan**

### CODE REVIEW

File: backend/canisters/identity/impl/src/model/identity_map.rs [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
+use types::TimestampMillis;
+use types::VerifiableField;
+use types::VerificationCodeStatus;
+use types::{FieldId, PhoneNumber};
 
 #[derive(Default)]
 pub struct IdentityMap {
     identities_by_principal: HashMap<Principal, Identity>,
-    // phone_number_to_principal: HashMap<PhoneNumber, Principal>,
+    registered_phone_numbers: HashSet<PhoneNumber>,
```
Comment (hpeebles) [inline backend/canisters/identity/impl/src/model/identity_map.rs:14]: We need a way of allowing phone numbers to be recycled. We could store the date each number is registered and allow them to be reused after 3 months or so.

File: backend/canisters/identity/impl/src/model/identity_map.rs [hunk 2]
```diff
@@ -18,4 +20,37 @@ impl IdentityMap {
...
+        let field = VerifiableField::<PhoneNumber> {
+            id,
+            status: VerificationCodeStatus::Pending,
+            added: now,
+            value: phone_number.clone(),
+        };
+
+        match self.identities_by_principal.entry(principal) {
+            Occupied(mut e) => {
+                let identity = e.get_mut();
```
Comment (hpeebles) [inline backend/canisters/identity/impl/src/model/identity_map.rs]: You should use `into_mut` here.
From my brief reading about the difference a while ago the general consensus was that you should use `into_mut` by default and only use `get_mut` if you need to reference the `OccupiedEntry` again later on.
`into_mut` will consume the `OccupiedEntry` so you don't need to mark it as `mut`.

File: backend/libraries/types/src/field_id.rs [hunk 3]
```diff
@@ -22,3 +23,9 @@ impl From<u128> for FieldId {
         FieldId(value)
     }
 }
+
+impl FieldId {
+    pub fn new() -> FieldId {
+        FieldId(Uuid::new_v4().as_u128())
```
Comment (hpeebles) [inline backend/libraries/types/src/field_id.rs:29]: Is this deterministic? I guess it must be...

File: backend/canisters/identity/impl/src/model/identity_map.rs (hunk 1 above)
Comment (megrogan) [reply backend/canisters/identity/impl/src/model/identity_map.rs:14]: Not a priority for the hackathon I suppose...

**Review by hpeebles — APPROVED** [review]

**Pull request merged**